tests/s3_bucket_policy_write.txt

```diff
@@ -15,9 +15,14 @@
             "Sid": "PublicObjectWrite",
             "Effect": "Allow",
             "Principal": "*",
-            "Action": [
-                "s3:PutObject"
-            ],
+            "Action": "s3:PutObject",
+            "Resource": "arn:aws:s3:::rms-filecache-tests-writable/*"
+        },
+        {
+            "Sid": "PublicObjectDelete",
+            "Effect": "Allow",
+            "Principal": "*",
+            "Action": "s3:DeleteObject",
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider consolidating similar S3 policy actions</b></div>
<div id="fix">

Consider consolidating the `s3:PutObject` and `s3:DeleteObject` actions into a single statement with an array of actions, similar to how `s3:GetObject` and `s3:GetObjectVersion` are structured. This would make the policy more maintainable and concise.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -14,14 +14,10 @@
          {
              "Sid": "PublicObjectWrite",
              "Effect": "Allow",
              "Principal": "*",
 -            "Action": "s3:PutObject",
 -            "Resource": "arn:aws:s3:::rms-filecache-tests-writable/*"
 -        },
 -        {
 -            "Sid": "PublicObjectDelete",
 -            "Effect": "Allow",
 -            "Principal": "*",
 -            "Action": "s3:DeleteObject",
 +            "Action": [
 +                "s3:PutObject",
 +                "s3:DeleteObject"
 +            ],
              "Resource": "arn:aws:s3:::rms-filecache-tests-writable/*"
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/SETI/rms-filecache/pull/32#issuecomment-2558670107>#9fe30b</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged